apps/minifront/src/components/staking/layout.tsx

```diff
@@ -1,25 +1,32 @@
 import { useEffect } from 'react';
-import { useStore } from '../../state';
-import { stakingSelector } from '../../state/staking';
+import { AllSlices, useStore } from '../../state';
 import { throwIfPraxNotConnectedTimeout } from '@penumbra-zone/client';
 import { Account } from './account';
+import { useStoreShallow } from '../../utils/use-store-shallow';
 
 export const StakingLoader = async () => {
   await throwIfPraxNotConnectedTimeout();
   // Await to avoid screen flicker.
-  await useStore.getState().staking.loadUnstakedAndUnbondingTokensByAccount();
+  await useStore.getState().staking.loadAndReduceBalances();
 
   return null;
 };
 
+const stakingLayoutSelector = (state: AllSlices) => ({
+  account: state.staking.account,
+  loadDelegationsForCurrentAccount: state.staking.loadDelegationsForCurrentAccount,
+  loadUnbondingTokensForCurrentAccount: state.staking.loadUnbondingTokensForCurrentAccount,
+});
+
 export const StakingLayout = () => {
-  const { account, loadDelegationsForCurrentAccount } = useStore(stakingSelector);
+  const { account, loadDelegationsForCurrentAccount, loadUnbondingTokensForCurrentAccount } =
+    useStoreShallow(stakingLayoutSelector);
 
   /** Load delegations every time the account changes. */
-  useEffect(
-    () => void loadDelegationsForCurrentAccount(),
-    [account, loadDelegationsForCurrentAccount],
-  );
+  useEffect(() => {
+    void loadDelegationsForCurrentAccount();
+    void loadUnbondingTokensForCurrentAccount();
```

Comment:
These seem to always be together in a number of places right? Should we just combine them into one `loadStakingPositions()` function?

Reply:
They don't all need to get called at the same time, so unfortunately, they need to be split this way. For example:
- `loadDelegationsForCurrentAccount` and `loadUnbondingTokensForCurrentAccount` get called here in the effect
- `loadAndReduceBalances` gets called in the loader
- `loadDelegationsForCurrentAccount` and `loadAndReduceBalances` get (re-)called after delegating (but there's no need to reload unbonding tokens at that point)
- All three get called when undelegating
- `loadAndReduceBalances` and `loadUnbondingTokensForCurrentAccount` get called after undelegate claim, since those pieces of data need to be reloaded